2.x-activity-module/mod/aspirelists/yui/build/moodle-mod_aspirelists-inline_display/moodle-mod_aspirelists-inline_display-debug.js

```diff
@@ -58,7 +65,91 @@ NS.resize_embedded_lists = function(e)
 
         o.setAttribute('width', width);
     });
-}
+};
 
+/**
+ * Handle expanded inline iFrames
+ *
+ * startQueue() will queue all Expanded inline iFrames to load sequentially,
+ * each one paced by the inline_display_delay time (milliseconds)
+ *
+ * loadIfIframeInViewport() will detect if a queued-but-not-yet-loaded resource
+ * is currently on screen.  If the user keeps it on screen for the inline_display_delay
+ * time (milliseconds) then the launch will be triggered.  Typically used
+ * as a handler for a scroll event.
+ *
+ * This will ignore any inline iframes which are
+ * currently collapsed.  These will be
+ * loaded when the user interacts with them.
+ *
+ * Call .init() to start the process
+ */
+NS.ExpandedInlineIFramesLoader = (function(){
+    var thisScope;
+    var loader = {
+        iframeQueue: [],
+        inline_display_delay: 1000,
+        viewport_hover_delay: 1000,
+        loadIfIframeInViewport: function(){
+            // A scroll has happened, so check every queued iFrame
+            // to see if it is visible, if so then wait for the value
+            // of viewport_hover_delay, and if the element is still visible
+            // then load it.
+            for (var i = 0; i < thisScope.iframeQueue.length; i++) {
+                var element = thisScope.iframeQueue[i]['element'];
+                if (Y.DOM.inViewportRegion(element.getDOMNode())) {
+                    var src = thisScope.iframeQueue[i]['src'];
+                    window.setTimeout((function (element, src, populateIFrameCallback) {
```

Comment:
It's not sleeping the code, simply asynchronously queueing up the next resource to load after after a short delay, as per the story requirement.

However, from external conversations it looks like this feature is being replaced with a new approach.